src/includes/class-alma-wc-refund.php

```diff
@@ -0,0 +1,239 @@
+
+		// @todo check if this message is the same for old WC versions.
+		if ( 'Order status set to refunded. To return funds to the customer you will need to issue a refund through your payment gateway.' === $comment_datas['comment_content'] ) {
+			/* translators: %s is a username. */
+			$comment_datas['comment_content'] = sprintf( __( 'Order fully refunded via Alma by %s.', 'alma-woocommerce-gateway' ), wp_get_current_user()->display_name );
+		}
+
+		return $comment_datas;
+	}
+
+	/**
+	 * WP action hook "current_screen".
+	 *
+	 * @return void
+	 */
+	public function woocommerce_order_item_add_action_buttons() {
+		if ( is_object( get_current_screen() ) && 'shop_order' === get_current_screen()->id ) {
+			add_filter( 'gettext', array( $this, 'gettext' ), 10, 3 );
+		}
+	}
+
+	/**
+	 * Filters the text of the refund button on a back-office order page.
+	 *
+	 * @param string $translation A text translated.
+	 * @param string $text A text to translate.
+	 * @param string $domain A text domain.
+	 * @return mixed|string
+	 */
+	public function gettext( $translation, $text, $domain ) {
+
+		if ( 'woocommerce' !== $domain || ! array_key_exists( $text, $this->admin_texts_to_change ) ) {
+			return $translation;
+		}
+
+		foreach ( $this->admin_texts_to_change as $original_text => $updated_text ) {
+			if ( $original_text === $text ) {
+				$order = wc_get_order( intval( $_GET['post'] ) ); // phpcs:ignore WordPress.Security.NonceVerification
+				if ( substr( $order->get_payment_method(), 0, 4 ) !== 'alma' ) {
+					return $translation;
+				}
+				$translation = str_replace( $original_text, $updated_text, $text );
+				$this->number_of_texts_changed++;
+			}
+		}
+
+		if ( count( $this->admin_texts_to_change ) === $this->number_of_texts_changed ) {
+			remove_filter( 'gettext', array( $this, 'gettext' ), 10 );
+		}
+
+		return $translation;
+	}
+
+	/**
+	 * Print refund notices.
+	 *
+	 * @return void
+	 */
+	public function admin_notices() {
+
+		if ( 'shop_order' !== get_current_screen()->id ) {
+			return;
+		}
+
+		global $post_id;
+		$refund_notices = get_post_meta( $post_id, 'alma_refund_notices', true );
+
+		if ( ! is_array( $refund_notices ) ) {
+			return;
+		}
+
+		foreach ( $refund_notices as $notice_infos ) {
+			if ( ! is_array( $notice_infos ) ) {
+				continue;
+			}
+			?>
+			<div class="notice notice-<?php echo esc_html( $notice_infos['notice_type'] ); ?> is-dismissible">
+				<p><?php echo esc_html( $notice_infos['message'] ); ?></p>
+			</div>
+			<?php
+		}
+		delete_post_meta( $post_id, 'alma_refund_notices' );
+	}
+
+	/**
+	 * Action hook for order partial refunded.
+	 *
+	 * @param integer $order_id Order id.
+	 * @param integer $refund_id Refund id.
+	 * @return void
+	 */
+	public function woocommerce_order_partially_refunded( $order_id, $refund_id ) {
+
+		$order = wc_get_order( $order_id );
+		if ( true !== $this->refund_helper->is_order_valid_for_partial_refund_with_alma( $order_id, $refund_id ) ) {
```

Comment:
```suggestion
		if ( ! $this->refund_helper->is_order_valid_for_partial_refund_with_alma( $order, $refund_id ) ) {
```

* `is_something` is already supposed to return a boolean, this is not necessary to compare it with `=== true`
* pass directly `$order` here instead of `$order_id`

Reply:
ok you were right. 
The method used order_id and not order, but the methods called inside this one had to do

Reply:
ok you were right.
The method used order_id and not order, but the methods called inside this one had to do
`$order = wc_get_order( $order_id );`
So I did like you asked to me, passing everytime `$order `and not `$order_id`.